ring-jetty-adapter/test/ring/adapter/test/jetty.clj

```diff
@@ -200,3 +201,38 @@
               (Thread/sleep 250)
               (recur (inc i))))
           (is (= thread-count (count (all-threads)))))))))
+
+(deftest test-ssl-context-factory
+  (defn- get-ssl-context-factory
+    [^Server s]
+    (->> (seq (.getConnectors s))
+         (map #(seq (.getConnectionFactories %)))
+         (flatten)
+         (filter #(instance? SslConnectionFactory %))
+         (first)
+         (.getSslContextFactory)))
+
+  (defn- with-ssl-server [app options f]
+    (let [options (conj options
+                        {:port test-port
```

Comment:
Factor this map out out into a `test-ssl-options` var, and `merge` is likely a better function to use than `join`.